bashbrew/bashbrew.sh

```diff
@@ -280,6 +278,22 @@ for repoTag in "${repos[@]}"; do
 	fi
 done
 
+# usage: gitCheckout "$gitRepo" "$gitRef" "$gitDir"
+gitCheckout() {
+	[ "$1" -a "$2" -a "$3" ] || return 1
+	(
+		set -x
+		cd "$1"
+		git reset -q HEAD
+		git checkout -q -- .
+		git clean -dfxq
+		git checkout -q "$2" --
+		cd "$1/$3"
+		"$dir/git-set-mtimes"
+	)
+	return 0
+}
```

Comment:
This one still won't "fail" properly, but at least it's abstracted for better readability.